### PR TITLE
#377 use JWT when making requests to the DAV server

### PR DESCRIPTION
**Pull request by rezk2ll**

### CODE REVIEW

**Comment by huy-ta** [comment]
You should rebase these 2 commits into 1.

**Review by renaudboyer — DISMISSED** [review]

**Review by huy-ta — DISMISSED** [review]

File: src/esn.calendar.libs/app/services/dav-request.js [hunk 1]
```diff
@@ -13,11 +13,15 @@ function calDavRequest($http, $q, tokenAPI, calCalDAVURLService, CAL_GRACE_DELAY
       delete params.graceperiod;
     }
 
-    return $q.all([tokenAPI.getNewToken(), _getDavServerUrl()])
-      .then(([{ data }, serverBaseUrl]) => {
+    return $q.all([tokenAPI.getNewToken(), _getDavServerUrl(), tokenAPI.getWebToken()])
```
Comment (huy-ta) [inline src/esn.calendar.libs/app/services/dav-request.js:16]: It bugs me to see that even though we're only using either ESNToken or JWT but we're sending requests to get both here... Why can't we just remove ESNToken? cc @renaudboyer @rezk2ll

Comment (rezk2ll) [reply src/esn.calendar.libs/app/services/dav-request.js:16]: it's to ease the transition from ESNTOKEN to JWT. if JWT is not available we use ESNTOKEN

Comment (huy-ta) [reply src/esn.calendar.libs/app/services/dav-request.js:16]: Why would JWT not be available though?

Comment (rezk2ll) [reply src/esn.calendar.libs/app/services/dav-request.js:16]: the public key can be missing from the server. the technical user making requests from the ESN backend etc...

Comment (huy-ta) [reply src/esn.calendar.libs/app/services/dav-request.js:16]: Okay

**Review by huy-ta — APPROVED** [review]

**Review by renaudboyer — APPROVED** [review]

**Pull request merged**